src/Components/Web.JS/src/Platform/Mono/MonoDebugger.ts

```diff
@@ -1,19 +1,32 @@
 import { WebAssemblyResourceLoader } from '../WebAssemblyResourceLoader';
 
+const BLAZOR_DEBUG_FLAG = '_blazor_debug';
+
 const currentBrowserIsChrome = (window as any).chrome
   && navigator.userAgent.indexOf('Edge') < 0; // Edge pretends to be Chrome
 
-let isDebugging = true;
+let isDebugging = sessionStorage.getItem(BLAZOR_DEBUG_FLAG);
 
 window.addEventListener('load', () => {
   const params = new URLSearchParams(window.location.search);
-  isDebugging = params.get('_blazor_debug') === 'true';
+  const isDebuggingParam = params.get(BLAZOR_DEBUG_FLAG);
+  // (1) Value from query param then (2) value in session storage then (3) default to true
+  isDebugging = isDebuggingParam !== null ? isDebuggingParam : isDebugging || 'true';
```

Comment:
Doesn't defaulting to true mean it's going to be true in typical production scenarios, where there's no query param and nothing in session storage?

Reply:
Not necessarily, because the `hasDebugging` check also checks to see if pdbs are generated which isn't the case in production scenarios.